tools/mwasmarm_patcher/mwasmarm_patcher.c

```diff
@@ -0,0 +1,112 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <openssl/sha.h>
+#include <string.h>
+#include <stdarg.h> 
+#include <stdlib.h>
+
+// mwasmarm patcher v1.1
+// Patches the Metrowerk C compiler assembler to stop the line ending bug.
+
+// Changelog:
+// v1.1: Added patch definitions and looped over them to find the matching
+// definition as well as the version.
+
+struct PatchDef {
+    char *version;
+    char *sha1before;
+    char *sha1after;
+    int offsetPatch;
+    int newByte;   
+};
+
+struct PatchDef gPatchDefs[] = {
+    // mwasmarm 1.2/base definition
+    {
+        "mwasmarm 1.2/base",
+        "87f942cc0a0e90e73550d8d6f3fffcdeb5f69fa5",
+        "2f1ccff22eaa443bb79235ca6477d3b86bdfd7e4",
+        0x57614,
+        0x5
+    },
+    // mwasmarm 2.0/base definition
+    {
+        "mwasmarm 2.0/base",
+        "9d63877c776245129b4727b41d3e9e63cfc9cd28",
+        "f5dea73bf90791e104cb59458bebae8b08a55484",
+        0x57644,
+        0x5
+    },
+    // mwasmarm 2.0/sp2p4 definition
+    {
+        "mwasmarm 2.0/sp2p4",
+        "448cb0c7f1ace4393e9a9562f819f7a9f049be83",
+        "c82161527277b991a1b77e14617a93bcd19cf95c",
+        0x57834,
+        0x5
+    },
+    {
+    }
+};
+
+void fatal_printf(char *str, ...) {
+    va_list args;
+    va_start(args, str);
+    vprintf(str, args);

```

Comment:
error message goes to stderr

Reply:
Not fixing. Standard fatal_printf implementation.